macros/edr/system/system_utils/buckets_cte.sql

```diff
@@ -1,75 +1,90 @@
 {% macro complete_buckets_cte(time_bucket) %}
-    {{ adapter.dispatch('complete_buckets_cte','elementary')(time_bucket) }}
+    {%- set edr_bucket_end_expr = elementary.edr_timeadd(time_bucket.period, time_bucket.count, 'edr_bucket_start') %}
+    {%- set edr_min_bucket_start_expr = elementary.edr_date_trunc(time_bucket.period, elementary.edr_cast_as_timestamp(elementary.edr_quote(elementary.get_min_bucket_start()))) %}
+    {%- set edr_max_bucket_end_expr = elementary.edr_cast_as_timestamp(elementary.edr_quote(elementary.get_max_bucket_end())) %}
+    {{ adapter.dispatch('complete_buckets_cte','elementary')(time_bucket, edr_bucket_end_expr, edr_min_bucket_start_expr, edr_max_bucket_end_expr) }}
 {% endmacro %}
 
-{% macro spark__complete_buckets_cte(time_bucket) %}
-    {% set edr_bucket_end_expr = elementary.edr_timeadd(time_bucket.period, time_bucket.count, 'edr_bucket_start') %}
+{% macro spark__complete_buckets_cte(time_bucket, edr_bucket_end_expr, edr_min_bucket_start_expr, edr_max_bucket_end_expr) %}
```

Comment:
really no need for the `edr_` prefix and the `_expr` suffix in the bucket_start, bucket_end, no?
Something like this would be cleaner?
```suggestion
{% macro spark__complete_buckets_cte(time_bucket, bucket_end, min_bucket_start, max_bucket_end) %}
```

Reply:
I kept the `edr_` as this was the convention before my changes. Can remove it probably.
`_exp` is important as these do not return a time, they return an SQL expression.